File: solutions/200/solution_200_test.py
"""200. Number of Islands tests."""
from solution_200 import Solution

TESTDATA = [
    [[["1", "1", "0", "0", "0"], ["1", "1", "0", "0", "0"],
      ["0", "0", "1", "0", "0"], ["0", "0", "0", "1", "1"]], 3],
    [[
        ["1"],
        ["1"],
        ["0"],
        ["1"],
        ["0"],
        ["1"],
        ["0"],
        ["1"],
        ["1"],
        ["1"],
    ], 4],
]


def test_solution():
    """Compares result of Solution with test input."""

    for test in TESTDATA:
        got, want = Solution().numIslands(test[0]), test[1]
        assert got == want

File: solutions/200/solution_200_test.go
package solution200

import "testing"

var (
	testData = [][]byte{
		[]byte("1"),
		[]byte("1"),
		[]byte("0"),
		[]byte("1"),
		[]byte("0"),
		[]byte("0"),
		[]byte("1"),
	}
	wantIslands = 3
)

func TestSolution(t *testing.T) {
	if got, want := numIslands(testData), wantIslands; got != want {
		t.Errorf("numIslands(%v): got number of islands: %d: want: %d", testData, got, want)
	}
}
